// python3 -m grpc_tools.protoc -I./proto --python_out=. --grpc_python_out=. ./proto/sink.proto
syntax = "proto3";
option java_multiple_files = true;
option java_package = "mliot.sensors.proto";

service SinkService {
  rpc onAudioFrameAvailable(AudioMessage) returns (Response) {}
  rpc onVideoFrameAvailable(VideoMessage) returns (Response) {}
  rpc onAccelerationChanged(AccelerationMessage) returns (Response) {}
  rpc setAccelerationMaxRange(RangeMessage) returns (Response) {}
  rpc onStepDetected(RangeMessage) returns (Response) {}
  rpc onMotionDetected(RangeMessage) returns (Response) {}
  rpc onTemperatureChanged(RangeMessage) returns (Response) {}
  rpc onHeartRateChanged(RangeMessage) returns (Response) {}
  rpc onProximityChanged(RangeMessage) returns (Response) {}
}

message Response {
  bool received = 1;
}

message AudioMessage {
  bytes audio_frame = 1;
}

message VideoMessage {
  bytes video_frame = 1;
}

message RangeMessage {
  float max_range = 1;
}

message AccelerationMessage {
  float x = 1;
  float y = 2;
  float z = 3;
}